Python/Product/ExternalProfilerDriver/VTuneStackParser.cs

```diff
@@ -143,4 +151,61 @@ static class VTuneStackParser {
         }
 
     }
+
+    class VTuneStackParserForCPP
+    {
+        public static IEnumerable<SampleWithTrace> ParseFromFile(string filename)
+        {
+            if (!File.Exists(filename)) {
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.ErrorMsgPathDoesNotExist, filename));
+            }
+
+            SampleWithTrace current = null;
+            List<PerformanceSample> currentStack = new List<PerformanceSample>();
+
+            uint index = 0;
+            using (var reader = new StreamReader(filename)) {
+                using (CsvReader csvReader = new CsvReader(reader)) {
+
+                    while (csvReader.Read()) {
+                        var row = csvReader.GetRecord<dynamic>();
+                        var dict = row as IDictionary<string, object>;
+
+                            if (dict["Function"] == null || dict["Function"].ToString() == string.Empty) {
```

Comment:
this is potentially going to lookup the same entry twice.

you could use (hopefully I have it right, I didn't compile this):
`if (!dict.TryGetValue("Function", out object funcObj) || string.IsNullOrEmpty(funcObj?.ToString()))`

Similar for Function Stack